packages/web-components/@groupby/elements-base/test/unit/base.test.ts

```diff
@@ -1,5 +1,86 @@
-import { expect, stub } from '../utils';
+import { expect, stub, spy } from '../utils';
 import { DummyComponent } from './dummy-component';
+import { dataInitializer } from '../../src';
+
+describe('dataInitializer decorator', () => {
+  const testPropertyName = 'results';
+  let testObj;
+
+  beforeEach(() => {
+    testObj = {
+      init: false,
+      get results() {
+        return this._results;
+      },
+      set results(val) {
+        this._results = val;
+      },
+      _results: ['a', 'b'],
+    };
+  });
+
+  it('should extend a property setter', () => {
+    const originalSet = spy(testObj, 'results', ['set']);
```

Comment:
Another approach is to have empty spies for the getter and setter instead of relying on the functionality of the underlying getter/setter. Have the original getter/setter call the spies and assert that the spies were called in the test.

Reply:
Updated to this pattern. Also involved no longer asserting a change in the `results` property on `testObj`, since technically that shouldn't be done anyways as it is testing something other than the decorator.